ospf6d/ospf6_network.c

```diff
@@ -113,7 +100,6 @@ int ospf6_serv_sock(struct ospf6 *ospf6)
 	ospf6_reset_mcastloop(ospf6_sock);
 	ospf6_set_pktinfo(ospf6_sock);
 	ospf6_set_transport_class(ospf6_sock);
-	ospf6_set_checksum(ospf6_sock);
```

Comment:
Why is this removed here?  How is this related to RFC7166?

Reply:
RFC7166 states that We need not have to calculate checksum if authentication trailer is enabled. So it needs to be turned on/off based on feature enablement. Have added internal function to do the same.